File: ios/Runner/Native/Display/BrightnessListener.swift
import UIKit
import Flutter

class BrightnessListener: NSObject, FlutterStreamHandler {
    private var eventSink: FlutterEventSink?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        startListening()
        
        // Send current brightness immediately
        let currentBrightness = Double(UIScreen.main.brightness)
        events(currentBrightness)
        
        print("🔆 Started brightness listening")
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        stopListening()
        self.eventSink = nil
        print("⏹️ Stopped brightness listening")
        return nil
    }
    
    private func startListening() {
        NotificationCenter.default.addObserver(
            self,
            selector: #selector(brightnessDidChange),
            name: UIScreen.brightnessDidChangeNotification,
            object: nil
        )
    }
    
    private func stopListening() {
        NotificationCenter.default.removeObserver(
            self,
            name: UIScreen.brightnessDidChangeNotification,
            object: nil
        )
    }
    
    @objc private func brightnessDidChange() {
        let brightness = Double(UIScreen.main.brightness)
        print("🔆 Brightness changed to: \(brightness)")
        eventSink?(brightness)
    }
}

File: android/app/src/main/kotlin/com/example/news_app/native/display/BrightnessListener.kt
import android.content.Context
import android.database.ContentObserver
import android.net.Uri
import android.os.Handler
import android.os.Looper
import android.provider.Settings
import io.flutter.plugin.common.EventChannel

class BrightnessListener(private val context: Context) : EventChannel.StreamHandler {
    private var eventSink: EventChannel.EventSink? = null
    private var brightnessObserver: ContentObserver? = null
    private val handler = Handler(Looper.getMainLooper())
    
    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
        startListening()
        
        // Send current brightness immediately
        try {
            val brightness = Settings.System.getInt(
                context.contentResolver,
                Settings.System.SCREEN_BRIGHTNESS
            ) / 255.0
            events?.success(brightness)
        } catch (e: Exception) {
            // Ignore
        }
    }
    
    override fun onCancel(arguments: Any?) {
        stopListening()
        eventSink = null
    }
    
    private fun startListening() {
        brightnessObserver = object : ContentObserver(handler) {
            override fun onChange(selfChange: Boolean, uri: Uri?) {
                super.onChange(selfChange, uri)
                try {
                    val brightness = Settings.System.getInt(
                        context.contentResolver,
                        Settings.System.SCREEN_BRIGHTNESS
                    ) / 255.0
                    eventSink?.success(brightness)
                } catch (e: Exception) {
                    // Ignore errors
                }
            }
        }
        
        try {
            context.contentResolver.registerContentObserver(
                Settings.System.getUriFor(Settings.System.SCREEN_BRIGHTNESS),
                false,
                brightnessObserver!!
            )
        } catch (e: Exception) {
            // Permission might not be available
        }
    }
    
    private fun stopListening() {
        brightnessObserver?.let {
            try {
                context.contentResolver.unregisterContentObserver(it)
            } catch (e: Exception) {
                // Observer already unregistered
            }
            brightnessObserver = null
        }
    }
    
    fun destroy() {
        stopListening()
    }
}
